### PR TITLE
[interp] Consolidate non-recursive call sites.

### PR DESCRIPTION
**Pull request by jaykrell**
[interp] Consolidate non-recursive call sites.
 Consolidate native_stack_addr as a result (resolving a FIXME).
 Builds on:
  https://github.com/mono/mono/pull/18857
  https://github.com/mono/mono/pull/18858
  https://github.com/mono/mono/pull/18869

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build

**Comment by jaykrell** [comment]
@monojenkins build

**Comment by jaykrell** [comment]
@monojenkins build

File: mono/mini/interp/interp.c [hunk 1]
```diff
@@ -3537,37 +3544,18 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
...
-#endif
-
-	MonoException *ex;
-
-	// FIXME Use method_entry here. But it assumes clause_args == NULL.
-
-	if (!frame->imethod->transformed) {
-#if DEBUG_INTERP
-		char *mn = mono_method_full_name (frame->imethod->method, TRUE);
-		g_print ("(%p) Transforming %s\n", mono_thread_internal_current (), mn);
```
Comment (lewurm) [inline mono/mini/interp/interp.c]: Why is this removed?

Comment (jaykrell) [reply mono/mini/interp/interp.c]: It is the same as method_entry. Use method_entry instead.
The switch from imethod to cmethod is part of the point -- that is passing a parameter across blocks.

Comment (jaykrell) [reply mono/mini/interp/interp.c]: The ifdef is how to minimize repetition.

**Comment by jaykrell** [comment]
https://github.com/mono/mono/pull/18857 is the first of a few small steps then.

**Comment by jaykrell** [comment]
https://github.com/mono/mono/pull/18858 is another small piece of this.

File: mono/mini/interp/interp.c (hunk 1 above)
Comment (jaykrell) [reply mono/mini/interp/interp.c]: If I may, I'm going to dismiss this.
1. We talked.
2. This PR is split up.
3. This part is in another PR, that is approved.
4. This PR will only have a little left.
5. It isn't approved yet anyway, dismissing this won't make it approved.
